src/main/java/com/github/hokkaydo/eplbot/module/code/RustCompiler.java

```diff
@@ -0,0 +1,129 @@
+package com.github.hokkaydo.eplbot.module.code;
+import java.io.ByteArrayOutputStream;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class RustCompiler implements Runner{
+    private static final String CURRENT_DIR = System.getProperty("user.dir") + File.separator+"src"+File.separator+"temp"+File.separator;
+    private static final String[] VALID_EXTENSIONS = {".exe", ".pdb", ".rs"};
+
+    private static final ScheduledExecutorService SCHEDULER = new ScheduledThreadPoolExecutor(1);
+    @Override
+    public String run(String input, Integer runTimeout) {
+        if (containsUnsafeKeywords(input)){
+            return "Compilation failed:\nCheck if 'std' or 'use' are used";
+        }
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        File sourceFile;
+        Process compile;
+        try {
+
+            sourceFile = new File(CURRENT_DIR, "temp.rs");
+            sourceFile.deleteOnExit();
+            FileWriter writer = new FileWriter(sourceFile);
+            writer.write(input);
+            writer.close();
+            compile = new ProcessBuilder("rustc", sourceFile.getAbsolutePath()).directory(new File(CURRENT_DIR)).redirectErrorStream(true).start();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+            return "Server error code R01" + e.toString();
```

Comment:
No need to give the stack trace to the user, the error code should be enough for us, devs, to identify the stack trace in the logs
```suggestion
            return "Server error code R01" + e;
```